azure/scope/machinepool.go

```diff
@@ -704,3 +717,39 @@ func (m *MachinePoolScope) UpdatePatchStatus(condition clusterv1.ConditionType,
 		conditions.MarkFalse(m.AzureMachinePool, condition, infrav1.FailedReason, clusterv1.ConditionSeverityError, "%s failed to update. err: %s", service, err.Error())
 	}
 }
+
+// PatchCAPIMachinePoolObject persists the capi machinepool configuration and status.
+func (m *MachinePoolScope) PatchCAPIMachinePoolObject(ctx context.Context) error {
+	return m.capiMachinePoolPatchHelper.Patch(
+		ctx,
+		m.MachinePool,
+	)
+}
+
+// UpdateCAPIMachinePoolReplicas updates the associated MachinePool replica count.
+func (m *MachinePoolScope) UpdateCAPIMachinePoolReplicas(ctx context.Context, replicas *int32) {
+	m.MachinePool.Spec.Replicas = replicas
+}
+
+// HasReplicasExternallyManaged returns true if the externally managed annotation is set on the CAPI MachinePool resource.
+func (m *MachinePoolScope) HasReplicasExternallyManaged(ctx context.Context) bool {
+	return m.MachinePool.Annotations[azure.ReplicasManagedByAutoscalerAnnotation] == "true"
```

Comment:
@jackfrancis do we need a PR to get rid of this CAPZ specific annotation in favor of the CAPI one that you added?

Reply:
we should even be able to get rid of this function altogether and directly use the `ReplicasManagedByExternalAutoscaler()` helper from CAPI in `ReconcileReplicas` below